lua/ide/components/bufferlist/component.lua

```diff
@@ -0,0 +1,247 @@
+		vim.api.nvim_buf_set_option(buf, "filetype", "bufferlist")
+		vim.api.nvim_buf_set_option(buf, "buftype", "nofile")
+		vim.api.nvim_buf_set_option(buf, "modifiable", false)
+		vim.api.nvim_buf_set_option(buf, "swapfile", false)
+		vim.api.nvim_buf_set_option(buf, "textwidth", 0)
+		vim.api.nvim_buf_set_option(buf, "wrapmargin", 0)
+
+		if not self.config.disable_keymaps then
+			vim.api.nvim_buf_set_keymap(buf, "n", self.config.keymaps.edit, "", {
+				silent = true,
+				callback = function()
+					self.open_buf({ fargs = {} })
+				end,
+			})
+			vim.api.nvim_buf_set_keymap(buf, "n", self.config.keymaps.edit_split, "", {
+				silent = true,
+				callback = function()
+					self.open_buf({ fargs = { "split" } })
+				end,
+			})
+			vim.api.nvim_buf_set_keymap(buf, "n", self.config.keymaps.edit_vsplit, "", {
+				silent = true,
+				callback = function()
+					self.open_buf({ fargs = { "vsplit" } })
+				end,
+			})
+			vim.api.nvim_buf_set_keymap(buf, "n", self.config.keymaps.close, "", {
+				silent = true,
+				callback = function()
+					self.close_buf()
+				end,
+			})
+		end
+
+		-- setup autocmds to refresh
+		vim.api.nvim_create_autocmd({ "BufAdd", "BufDelete", "BufWipeout" }, {
+			callback = self.refresh,
+			group = vim.api.nvim_create_augroup("NvimIdeBufferlist", { clear = true }),
+		})
+
+		return buf
+	end
+
+	function self.open()
+		local log = self.logger.logger_from(nil, "Component.open")
+		log.debug("BufferList component opening, workspace %s", vim.api.nvim_get_current_tabpage())
+
+		-- init buf if not already
+		if self.buf == nil then
+			log.debug("buffer does not exist, creating.", vim.api.nvim_get_current_tabpage())
+			self.buf = setup_buffer()
+		end
+
+		log.debug("using buffer %d", self.buf)
+
+		-- initial render
+		self.refresh()
+		return self.buf
+	end
+
+	function self.post_win_create()
+		local log = self.logger.logger_from(nil, "Component.post_win_create")
+		-- setup web-dev-icons highlights if available
+		if pcall(require, "nvim-web-devicons") then
+			for _, icon_data in pairs(require("nvim-web-devicons").get_icons()) do
+				local hl = "DevIcon" .. icon_data.name
+				vim.cmd(string.format("syn match %s /%s/", hl, icon_data.icon))
+			end
+		end
+		-- set highlights for global icon theme
+		icon_set.set_win_highlights()
+		libwin.set_winbar_title(0, "BUFFERS")
+	end
+
+	function self.get_commands()
+		return commands.new(self).get()
+	end
+
+	function self.refresh()
+		local bufs = vim.tbl_map(function(buf)
+			local icon
+			-- use webdev icons if possible
+			if pcall(require, "nvim-web-devicons") then
+				local name = vim.api.nvim_buf_get_name(buf)
+				local ext = vim.fn.fnamemodify(name, ":e:e")
+				icon = require("nvim-web-devicons").get_icon(name, ext, { default = true })
+				if self.kind == "dir" then
+					icon = require("nvim-web-devicons").get_icon("dir")
+				end
+			else
+				if self.kind ~= "dir" then
+					icon = require("ide.icons").global_icon_set.get_icon("File")
```

Comment:
As a rule of thumb, if I call an inline "require" more then once, I make it a proper import at top of file. Just a nit.